public/app/core/components/TraceToProfiles/TraceToProfilesSettings.tsx

```diff
@@ -0,0 +1,222 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useAsync } from 'react-use';
+
+import {
+  DataSourceJsonData,
+  DataSourceInstanceSettings,
+  DataSourcePluginOptionsEditorProps,
+  updateDatasourcePluginJsonDataOption,
+} from '@grafana/data';
+import { ConfigSection } from '@grafana/experimental';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { InlineField, InlineFieldRow, Input, InlineSwitch } from '@grafana/ui';
+import { ConfigDescriptionLink } from 'app/core/components/ConfigDescriptionLink';
+import { DataSourcePicker } from 'app/features/datasources/components/picker/DataSourcePicker';
+import { ProfileTypesCascader } from 'app/plugins/datasource/grafana-pyroscope-datasource/QueryEditor/ProfileTypesCascader';
+import { PyroscopeDataSource } from 'app/plugins/datasource/grafana-pyroscope-datasource/datasource';
+import { ProfileTypeMessage } from 'app/plugins/datasource/grafana-pyroscope-datasource/types';
+
+import { IntervalInput } from '../IntervalInput/IntervalInput';
+import { TagMappingInput } from '../TraceToLogs/TagMappingInput';
+import { getTimeShiftLabel, getTimeShiftTooltip, invalidTimeShiftError } from '../TraceToLogs/TraceToLogsSettings';
+
+export interface TraceToProfilesOptions {
+  datasourceUid?: string;
+  tags?: Array<{ key: string; value?: string }>;
+  spanStartTimeShift?: string;
```

Comment:
Do we need this? If we use spanID we don't really need to be precise with the time so we don't need user to really tinker with that.

Reply:
Yes be we can easily just have span time + 1h or what ever. It just does not seem like it should be necessary for the user to specify it in this case. In logs we have it because there is valid usecase for searching for logs without traceID or spanID.